libimagref/src/reference.rs

```
@@ -230,11 +231,18 @@ impl<'a> Ref<'a> {
 
     /// Get the hash from the path of the ref
     pub fn get_path_hash(&self) -> Option<String> {
-        let pb : PathBuf = self.0.get_location().clone().into();
-        pb.file_name()
-            .and_then(|osstr| osstr.to_str())
-            .and_then(|s| s.split("~").next())
-            .map(String::from)
+        self.0
+            .get_location()
+            .clone()
+            .into_pathbuf()
+            .map_err_trace()
+            .ok() // TODO: Hiding the error here is not so nice
```

If `None` is returned in this specific case, then hiding the Error shouldnt be that big a deal. Otherwise, you'd have to change the return type to `Result<Option<String>>'